src/hooks/transfers/useTransfers.tsx

```diff
@@ -0,0 +1,276 @@
+  getNobleChainId,
+  getOsmosisChainId,
+  NEUTRON_BECH32_PREFIX,
+  OSMO_BECH32_PREFIX,
+} from '@/constants/graz';
+import {
+  COSMOS_SWAP_VENUES,
+  getDefaultChainIDFromNetworkType,
+  getDefaultTokenDenomFromAssets,
+  getNetworkTypeFromWalletInfo,
+  SWAP_VENUES,
+  TransferType,
+} from '@/constants/transfers';
+import { DydxAddress, EvmAddress, SolAddress } from '@/constants/wallets';
+
+import { getSelectedDydxChainId } from '@/state/appSelectors';
+import { useAppSelector } from '@/state/appTypes';
+
+import { convertBech32Address } from '@/lib/addressUtils';
+import { isNativeDenom } from '@/lib/assetUtils';
+
+import { useAccounts } from '../useAccounts';
+import { useSkipClient } from './skipClient';
+
+export const useTransfers = () => {
+  const { skipClient } = useSkipClient();
+  const { dydxAddress, sourceAccount } = useAccounts();
+  const selectedDydxChainId = useAppSelector(getSelectedDydxChainId);
+
+  const [fromTokenDenom, setFromTokenDenom] = useState<string | null>(null);
+  const [fromChainId, setFromChainId] = useState<string | null>(null);
+  const [toTokenDenom, setToTokenDenom] = useState<string | null>(null);
+  const [toChainId, setToChainId] = useState<string | null>(null);
+  const [fromAddress, setFromAddress] = useState<EvmAddress | SolAddress | DydxAddress | undefined>(
+    undefined
+  );
+  const [toAddress, setToAddress] = useState<EvmAddress | SolAddress | DydxAddress | undefined>(
+    undefined
+  );
+  const [transferType, setTransferType] = useState<TransferType>(TransferType.Withdraw);
+  const [amount, setAmount] = useState<string>('');
+
+  const chainsQuery = useQuery({
+    queryKey: ['transferEligibleChains'],
+    queryFn: async () => {
+      const skipSupportedChains = await skipClient.chains({
+        includeEVM: true,
+        includeSVM: true,
+      });
+      const chainsByNetworkMap = skipSupportedChains.reduce<Record<string, Chain[]>>(
+        (chainsMap, nextChain) => {
+          const chainsListForNetworkType = chainsMap[nextChain.chainType] ?? [];
+          chainsMap[nextChain.chainType] = [...chainsListForNetworkType, nextChain];
+          return chainsMap;
+        },
+        {}
+      );
+      return {
+        skipSupportedChains,
+        chainsByNetworkMap,
+      };
+    },
+    refetchOnWindowFocus: false,
+    refetchOnMount: false,
+    refetchOnReconnect: false,
+  });
+  const assetsQuery = useQuery({
+    queryKey: ['transferEligibleAssets'],
+    queryFn: async () => {
+      const assetsByChain = await skipClient.assets({
+        includeEvmAssets: true,
+        includeSvmAssets: true,
+      });
+      return { assetsByChain };
+    },
+    refetchOnWindowFocus: false,
+    refetchOnMount: false,
+    refetchOnReconnect: false,
+  });
+  const { chainsByNetworkMap = {} } = chainsQuery.data ?? {};
+  const { assetsByChain = {} } = assetsQuery.data ?? {};
+
+  const walletNetworkType = getNetworkTypeFromWalletInfo(sourceAccount?.walletInfo);
```

Comment:
sg! although the function will still have to exist, since we need to return `svm` instead of `solana`. but yeah the input type as `sourceAccount.chain` will def be more accurate

Reply:
you can just change both to 'svm'

Reply:
down to do in separate PR - that enum touches a bunch of files and would be nice to keep changes clean as possible